internal/runtimehandlerhooks/utils.go

```diff
@@ -121,3 +124,82 @@ func UpdateIRQSmpAffinityMask(cpus, current string, set bool) (cpuMask, bannedCP
 	}
 	return maskStringWithComma, invertedMaskStringWithComma, nil
 }
+
+// cpuMaskToCPUSet parses a CPUSet received in a Mask Format, see:
+// https://man7.org/linux/man-pages/man7/cpuset.7.html#FORMATS
+func cpuMaskToCPUSet(cpuMask string) (cpuset.CPUSet, error) {
+	chunks := strings.Split(cpuMask, ",")
+
+	// reverse the chunks order
+	n := len(chunks)
+	for i := 0; i < n/2; i++ {
+		chunks[i], chunks[n-i-1] = chunks[n-i-1], chunks[i]
+	}
+
+	builder := cpuset.NewBuilder()
+	for i, chunk := range chunks {
+		mask, err := strconv.ParseUint(chunk, 16, 32)
+		if err != nil {
+			return cpuset.NewCPUSet(), fmt.Errorf("failed to parse the CPU mask %s: %v", cpuMask, err)
+		}
+		for j := 0; j < 32; j++ {
+			if mask&1 == 1 {
+				builder.Add(i*32 + j)
+			}
+			mask >>= 1
+		}
+	}
+
+	return builder.Result(), nil
+}
+
+// cpuNumToCPUMask converts a number of CPUs to the corresponding CPU mask.
+// For example 18 becomes 3ffff. Note the commas are omitted, no need
+// since the functions is for internal use only
+func cpuNumToCPUMask(ncpus int) string {
+	var builder strings.Builder
+
+	for i := 0; i < ncpus-ncpus%4; i += 4 {
+		builder.WriteString("f")
+	}
+
+	switch ncpus % 4 {
+	case 1:
+		builder.WriteString("1")
+	case 2:
+		builder.WriteString("3")
+	case 3:
+		builder.WriteString("7")
+	}
+
+	mask := builder.String()
+	slice := []byte(mask)
+
+	// reverse
+	n := len(slice)
+	for i := 0; i < n/2; i++ {
+		slice[i], slice[n-i-1] = slice[n-i-1], slice[i]
+	}
+
+	return string(slice)
+}
+
+// machineCPUsCount returns the number of processing units available
+func machineCPUsCount() (int, error) {
```

Comment:
This is only used to create a long enough bitmask (allocating space and filling it with 1s). Executing an external tool for that seems to be a bit excessive. Is there a better / cheaper way? Reading /proc or /sys or maybe just using a big enough number (even though I would prefer something dynamic)?

Reply:
Indeed, as stated above reading /sys/devices/system/cpu/online will make this function obsolete